common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -7303,6 +7304,12 @@ public Map<String, String> subtree(String string) {
     return ret;
   }
 
+  public static boolean shouldComputeLineage(HiveConf conf) {
+    Collection<String> lineage_filter =
+      conf.getTrimmedStringCollection(HiveConf.ConfVars.HIVE_LINEAGE_STATEMENT_FILTER.varname);
+    return !(lineage_filter.isEmpty() || lineage_filter.contains("NONE"));
```

Comment:
I think HIVE_LINEAGE_INFO also has to be checked like
```
return !(lineage_filter.isEmpty() || lineage_filter.contains("NONE")) ||
  conf.getBoolVar(HiveConf.ConfVars.HIVE_LINEAGE_INFO);
```
Or it should be removed completely since it is deprecated.
Maybe we should keep it in the next Hive release but remove it in a follow-up patch for the 2nd release.

Reply:
Agree with that. It is deprecated, not removed. So it should still work.

Reply:
Ack